x-pack/plugins/security/server/authentication/providers/basic.ts

```diff
@@ -100,8 +100,15 @@ export class BasicAuthenticationProvider extends BaseAuthenticationProvider {
   /**
    * Redirects user to the login page preserving query string parameters.
    * @param request Request instance.
+   * @param [state] Optional state object associated with the provider.
    */
-  public async logout(request: KibanaRequest) {
+  public async logout(request: KibanaRequest, state?: ProviderState | null) {
+    this.logger.debug(`Trying to log user out via ${request.url.path}.`);
+
+    if (!state) {
```

Comment:
__note:__ need this check here and in other providers since I can call `logout` on multiple providers if there is no session (e.g. in case IdP initiated logout).